src/main/java/org/corfudb/infrastructure/PeriodicPollPolicy.java

```diff
@@ -0,0 +1,182 @@
+import org.corfudb.runtime.clients.IClientRouter;
+import org.corfudb.runtime.view.Layout;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Simple Polling policy.
+ * Every node polls every other node.
+ * <p>
+ * Failure Condition:
+ * Considers failure if server a node does not respond to the ping
+ * more than 2 times in a row.
+ * <p>
+ * Created by zlokhandwala on 9/29/16.
+ */
+@Slf4j
+public class PeriodicPollPolicy implements IFailureDetectorPolicy {
+
+    private CorfuRuntime corfuRuntime;
+    /**
+     * list of layout servers that we monitor.
+     */
+    private String[] historyServers = null;
+    private IClientRouter[] historyRouters = null;
+
+    /**
+     * polling history
+     */
+    private int[] historyPollFailures = null;
+    private int historyPollCount = 0;
+    private HashMap<String, Boolean> historyStatus = null;
+
+    /**
+     * Retry timeouts
+     */
+    long timeoutConnect = 50;
+    long timeoutRetry = 200;
+    long timeoutResponse = 1000;
+
+    /**
+     * Failed Poll Limit.
+     * Failed pings after which a node is considered dead.
+     */
+    long failedPollLimit = 3;
+
+    /**
+     * Executes the policy once.
+     * Checks for changes in the layout.
+     * Then polls all the servers and updates the status.
+     *
+     * @param layout Current Layout
+     */
+    @Override
+    public void executePolicy(Layout layout, CorfuRuntime corfuRuntime) {
+
+        this.corfuRuntime = corfuRuntime;
+        String[] allServers = layout.getAllServers().stream().toArray(String[]::new);
+        // Performs setup and checks for changes in the layout to update failure count
+        checkForChanges(allServers);
+        // Perform polling of all servers in historyServers
+        pollOnce();
+
+    }
+
+    /**
+     * Checks for changes in the layout.
+     * Fetches corfu runtime router to be able to ping a node.
+     *
+     * @param allServers List of all servers in teh layout
+     */
+    private void checkForChanges(String[] allServers) {
+
+        Arrays.sort(allServers);
+
+        if (historyServers == null || !Arrays.equals(historyServers, allServers)) {
+            // Initialize the status map for the first time or if there is a change.
+            if (historyStatus == null) {
+                historyStatus = new HashMap<>();
+            }
+
+            log.debug("historyServers change, length = {}", allServers.length);
+            historyServers = allServers;
+            historyRouters = new IClientRouter[allServers.length];
+            historyPollFailures = new int[allServers.length];
+            for (int i = 0; i < allServers.length; i++) {
+                if (!historyStatus.containsKey(allServers[i])) {
+                    historyStatus.put(allServers[i], true);  // Assume it's up until we think it isn't.
+                }
+                historyRouters[i] = corfuRuntime.getRouterFunction.apply(allServers[i]);
```

Comment:
Yes you are right, there is no other use for the runtime. Ill remove the field and pass it as a parameter.